modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/web/internal/portlet/action/test/GetWorkflowDefinitionInfoMVCResourceCommandTest.java

```diff
@@ -0,0 +1,143 @@
+import com.liferay.portal.workflow.manager.WorkflowDefinitionManager;
+import com.liferay.portal.workflow.metrics.service.util.WorkflowDefinitionUtil;
+
+import java.io.ByteArrayOutputStream;
+
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Pedro Leite
+ */
+@RunWith(Arquillian.class)
+public class GetWorkflowDefinitionInfoMVCResourceCommandTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_user = UserTestUtil.addUser();
+
+		_permissionChecker = PermissionCheckerFactoryUtil.create(_user);
+
+		_workflowDefinition =
+			_workflowDefinitionManager.deployWorkflowDefinition(
+				null, TestPropsValues.getCompanyId(),
+				TestPropsValues.getUserId(), StringUtil.randomId(),
+				StringUtil.randomId(), WorkflowDefinitionUtil.getBytes());
+	}
+
+	@Test
+	public void testGetWorkflowDefinitionInfo() throws Exception {
+		MockLiferayResourceRequest mockLiferayResourceRequest =
+			new MockLiferayResourceRequest();
+
+		mockLiferayResourceRequest.setAttribute(WebKeys.LOCALE, LocaleUtil.US);
+
+		ThemeDisplay themeDisplay = new ThemeDisplay();
+
+		themeDisplay.setCompany(
+			_companyLocalService.getCompany(TestPropsValues.getCompanyId()));
+		themeDisplay.setPermissionChecker(_permissionChecker);
+		themeDisplay.setUser(_user);
+
+		mockLiferayResourceRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, themeDisplay);
+
+		mockLiferayResourceRequest.setParameter(
+			"workflowDefinitionName", _workflowDefinition.getName());
+		mockLiferayResourceRequest.setParameter(
+			"workflowDefinitionVersion",
+			String.valueOf(_workflowDefinition.getVersion()));
+
+		MockLiferayResourceResponse mockLiferayResourceResponse =
+			new MockLiferayResourceResponse();
+
+		_mvcResourceCommand.serveResource(
+			mockLiferayResourceRequest, mockLiferayResourceResponse);
+
+		ByteArrayOutputStream byteArrayOutputStream =
+			(ByteArrayOutputStream)
+				mockLiferayResourceResponse.getPortletOutputStream();
+
+		JSONObject jsonObject = JSONFactoryUtil.createJSONObject(
+			byteArrayOutputStream.toString());
+
+		Assert.assertNotNull(jsonObject);
+
+		List<WorkflowNode> workflowNodes =
+			_workflowDefinition.getWorkflowNodes();
+
+		JSONArray workflowNodesJSONArray = jsonObject.getJSONArray("nodes");
+
+		Assert.assertEquals(
+			workflowNodes.size(), workflowNodesJSONArray.length());
+
+		List<WorkflowTransition> workflowTransitions =
+			_workflowDefinition.getWorkflowTransitions();
+
+		JSONArray workflowtransitionsJSONArray = jsonObject.getJSONArray(
+			"transitions");
+
+		Assert.assertEquals(
+			workflowTransitions.size(), workflowtransitionsJSONArray.length());
```

Comment:
@pedro-oliveira446 instead of asserting only the size, please try to use the `JSONAssert.assertEquals` to validate the entire JSON